src/Datadog.Trace/TraceContext.cs

```diff
@@ -76,6 +76,12 @@ public void AddSpan(Span span)
                                 Tracer.Sampler?.GetSamplingPriority(RootSpan);
                         }
                     }
+
+                    // set the origin tag to the root span of each trace/subtrace
+                    if (span.Context.Origin != null)
```

Comment:
Maybe it would make sense to move that to DecorateRootSpan? Currently it only has Azure stuff, but given the name of the method I feel like it would be semantically appropriate

Reply:
Sure, I saw the method but as you said, it has Azure stuff, I'll move this block to `DecorateRootSpan`